custom_components/homewhiz/api.py

```diff
@@ -213,54 +217,80 @@ async def make_api_get_request(
                 raise RequestError(contents)
 
 
-async def fetch_appliance_contents(
-    credentials: LoginResponse, app_id: str
-) -> ApplianceContents:
-    contents_index_response = await make_api_get_request(
+async def fetch_contents_index(
+    credentials: LoginResponse, app_id: str, language="en-GB"
+):
+    response = await make_api_get_request(
         host="api.arcelikiot.com",
         canonical_uri="/procam/contents",
         canonical_querystring=(
             f"applianceId={app_id}&"
             f"ctype=CONFIGURATION%2CLOCALIZATION&"
-            f"lang=en-GB&testMode=true"
+            f"lang={language}&"
+            f"testMode=true"
         ),
         credentials=credentials,
     )
-    contentsIndex = from_dict(ContentsIndexResponse, contents_index_response["data"])
-    config_contents = [
-        content for content in contentsIndex.results if content.ctype == "CONFIGURATION"
-    ]
+    return from_dict(ContentsIndexResponse, response["data"])
+
+
+async def fetch_base_contents_index(credentials: LoginResponse, language: str):
```

Comment:
Doesn't really matter but `fetch_appliance_contents` has a default language while this function doesn't.

Reply:
I actually removed it on purpose, as this one's purpose is to fetch only the localizations and should never be called without the language. Maybe I should rename it?

Reply:
No, makes sense!